database/manager.go

```diff
@@ -79,178 +71,32 @@ func (dm *DatabaseManager) Run() {
 				continue
 			}
 		}
+		dm.committedHeightsLock.RUnlock()
 	}
 }
 
-func (dm *DatabaseManager) Get(id RelayerID, key DataKey) ([]byte, error) {
-	return dm.db.Get(id.ID, key)
-}
-
-func (dm *DatabaseManager) PrepareHeight(id RelayerID, height uint64, totalMessages uint64) error {
-	km, ok := dm.keyManagers[id]
-	if !ok {
-		dm.logger.Error("Key manager not found", zap.String("relayerID", id.ID.String()))
-		return fmt.Errorf("key manager not found")
+func (dm *DatabaseManager) RegisterRelayerID(id RelayerID) {
+	dm.committedHeightsLock.Lock()
+	defer dm.committedHeightsLock.Unlock()
+	if _, ok := dm.committedHeights[id]; !ok {
+		dm.committedHeights[id] = 0
 	}
-	km.prepareHeight(height, totalMessages)
-	return nil
 }
 
-func (dm *DatabaseManager) Finished(id RelayerID, height uint64) {
-	km, ok := dm.keyManagers[id]
-	if !ok {
-		dm.logger.Error("Key manager not found", zap.String("relayerID", id.ID.String()))
-		return
-	}
-	km.finished <- height
-}
-
-// intHeap adapted from https://pkg.go.dev/container/heap#example-package-IntHeap
-type intHeap []uint64
-
-func (h intHeap) Len() int           { return len(h) }
-func (h intHeap) Less(i, j int) bool { return h[i] < h[j] }
-func (h intHeap) Swap(i, j int)      { h[i], h[j] = h[j], h[i] }
-func (h intHeap) Peek() uint64       { return h[0] }
-
-func (h *intHeap) Push(x any) {
-	// Push and Pop use pointer receivers because they modify the slice's length,
-	// not just its contents.
-	*h = append(*h, x.(uint64))
+func (dm *DatabaseManager) CommitHeight(id RelayerID, height uint64) {
+	dm.committedHeightsLock.Lock()
+	defer dm.committedHeightsLock.Unlock()
+	dm.committedHeights[id] = height
```

Comment:
Bear with me... But I don't think we should grab the lock here. I think it's okay to lock in `RegisterRelayerID`, and then grabbing a read lock while iterating over them so that the map doesn't get any new keys while iterating, but I think in this function, where we should only be updating the value for existing keys, we don't need to lock. Right now, this function will be locked out during the entire execution of the update tick when I don't think that serves any purpose.